File: pset6/crack.py
import crypt
import sys
import time

time_process = 0.0

if len(sys.argv) != 2:
    print("Usage: python crack.py hashed_password")
    exit(1)

salt = str(50)
before = time.process_time()
def main():
    for i in range(65, 65+26+26):
        if i<91:
            w = chr(i)
        else:
            w = chr(i+6)
        for j in range(65, 65+26+26):
            if j<91:
                o = chr(j)
            else:
                o = chr(j+6)
            for k in range(65, 65+26+26):
                if k<91:
                    r = chr(k)
                else:
                    r = chr(k+6)
                for l in range(65, 65+26+26):
                    if l<91:
                        d = chr(l)
                    else:
                        d = chr(l+6)
                    if check(w, o, r, d):
                        after = time.process_time()
                        time_process = after - before
                        print("Time program took while cracking the password = {:.2f} sec.".format(time_process))
                        exit (0)
                    else:
                        continue

    
def check(w, o, r, d):
    word = w+o+r+d
    hash1 = crypt.crypt(word[0:4], salt)
    if hash1 == sys.argv[1]:
        print(word)
        return True
    
    hash2 = crypt.crypt(word[0:3], salt)
    if hash2 == sys.argv[1]:
        print(word[0:3])
        return True
    
    hash3 = crypt.crypt(word[0:2], salt)
    if hash3 == sys.argv[1]:
        print(word[0:2])
        return True

    hash4 = crypt.crypt(word[0:1], salt)
    if hash4 == sys.argv[1]:
        print(word[0:1])
        return True
    return False
before1 = time.process_time()
if __name__ == "__main__":
    main()

File: pset2/crack.c
#include<cs50.h>
#include<stdio.h>
#include<string.h>
#define _XOPEN_SOURCE
#include<unistd.h>

char *crypt(const char *key, const char *salt);
int check(char* , char* , char*);
int main(int argc, char* argv[])
{
    if(argc != 2)
    {
        printf("Usage: ./crack hash\n");
        return 1;
    }
    char s[2];
    s[0] = argv[1][0];
    s[1] = argv[1][1];
    char* hash = argv[1];
    char pass[5];
    for(int i=65;i < 91 ; i++)
    {
        pass[0] = i;
        for(int j =1 ;j< 5;j++)
        pass[j] = '\0';
        if(!check(pass, s, hash))
        {
            return 0;
        }
        pass[0] = i + 32;
        if(!check(pass, s, hash))
        {
            return 0;
        }
    }
    return 0;
}
int check(char* pass, char* s, char* hash)
{
    char* result = crypt(pass, s);
    int c = 0,n = strlen(result), i;
    for(i=0; i < n; i++)
    {
        if(hash[i] == result[i])
        {
            c++;
        }
    }
    if(c == n)
    {
        printf("%s", pass);
        printf("\n");
        return 0;
    }
return 2;
}